host-controller/src/main/java/org/jboss/as/host/controller/operations/ServerStartHandler.java

```diff
@@ -108,16 +108,17 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 final ServerStatus origStatus = serverInventory.determineServerStatus(serverName);
                 if (origStatus != ServerStatus.STARTED && origStatus != ServerStatus.STARTING) {
                     final ServerStatus status = serverInventory.startServer(serverName, model, blocking);
+                    context.readResource(PathAddress.EMPTY_ADDRESS, false);
```

Comment:
AIUI you call this to trigger a call to ServerConfigResource.persistAutoStart()? If so you'd need to add a getModel() call. Also, there should be a comment explaining what this is about, or someone will likely delete it.

Same comment applies to other similar spots; e.g. L121 below.